src/interpreter/mod.rs

```diff
@@ -1081,12 +1090,12 @@ mod tests {
                 inner: secp256k1::PublicKey::from_secret_key(&secp, &sk),
                 compressed: true,
             };
-            let sig = secp.sign_ecdsa(&msg, &sk);
+            let signature = secp.sign_ecdsa(&msg, &sk);
             ecdsa_sigs.push(bitcoin::ecdsa::Signature {
-                sig,
-                hash_ty: bitcoin::sighash::EcdsaSighashType::All,
+                signature,
+                sighash_type: bitcoin::sighash::EcdsaSighashType::All,
             });
-            let mut sigser = sig.serialize_der().to_vec();
+            let mut sigser = signature.serialize_der().to_vec();
```

Comment:
Can't we do `serialize` trick here to avoid one heap allocation?
Also aren't we intending to deprecate `to_vec` anyways?

Reply:
#687.